project/main.py

```diff
@@ -54,11 +56,22 @@ def get_task_status(id):
 
 @app.get("/task/{id}/result")
 def get_schedule_result(id):
-    # return StreamingResponse(iterfile(f'{id}_rostering.json'), media_type = "application/json")
+    if not task_exists(id):
+        return JSONResponse(status_code=404)
+
     return StreamingResponse(iterfile(f'./tmp/{id}/rostering.json'), media_type = "application/octet-stream")
 
 
 @app.get("/task/{id}/statistics-results")
 def get_stats_result(id):
+    if not task_exists(id):
+        return JSONResponse(status_code=404)
     return StreamingResponse(iterfile(f'./tmp/{id}/statistics_output.json'), media_type="application/octet-stream")
 
+@app.get("/task/{id}/cancel")
+def cancel_task(id):
+    if not task_exists(id):
```

Comment:
should validate by celery whether task exists without file structure
its theoretically possible the directory does not exist, but celery runs task in background, hence we shall stop that task

Reply:
celery task owns dir creation on start here https://github.com/gitpay-ru/workforce-mvp/blob/master/project/worker.py#L102

Reply:
*if* there is a dedicated straigthforward api to check task status - its better to use that without 3rd statuses, 
consider using https://docs.celeryq.dev/en/stable/userguide/tasks.html#built-in-states if applicable